### DISCUSSION TITLE
Can't I use suspense client side only?

### DISCUSSION THREAD

**Discussion by testerez** [trpc/trpc#4825] [Q&A]
I'm trying to use suspense on the client side, but as soon as I set `{suspense: true}` on a query, it gets fetched server side, even though I've set `ssr: false` in `createTRPCNext`.
I can't find any clear documentation about this. Is it the expected behaviour?
And is there any way to use suspense without server side fetching?

I've reproduce the issue in a [sample project](https://codesandbox.io/p/github/testerez/trpc-suspense-experiment/main?file=%2Fsrc%2Fpages%2Findex.tsx%3A1%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clmugfcgo00073b6mc8avrgyd%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clmugfcgo00033b6mrzrzlgoi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clmugfcgo00053b6mhtrj4tka%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clmugfcgo00063b6mmidqunxm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clmugfcgo00033b6mrzrzlgoi%2522%253A%257B%2522id%2522%253A%2522clmugfcgo00033b6mrzrzlgoi%2522%252C%2522activeTabId%2522%253A%2522clmuzcgyt00033b6n12hlt7ht%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clmugfcgn00023b6mrz21o1m7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Futils%252Ftrpc.ts%2522%252C%2522id%2522%253A%2522clmupsmlk005l3b6njck3op2c%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clmuzcgyt00033b6n12hlt7ht%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fpages%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clmugfcgo00063b6mmidqunxm%2522%253A%257B%2522id%2522%253A%2522clmugfcgo00063b6mmidqunxm%2522%252C%2522activeTabId%2522%253A%2522clmugft0300793b6mvxfnzb6b%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clmugft0300793b6mvxfnzb6b%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clmugfcgo00053b6mhtrj4tka%2522%253A%257B%2522id%2522%253A%2522clmugfcgo00053b6mhtrj4tka%2522%252C%2522activeTabId%2522%253A%2522clmugfpuw005w3b6mvrsw4q2q%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clmugfcgo00043b6mixezwcc3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clmugfcla000ye4gs5tnrh17y%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clmugfpuw005w3b6mvrsw4q2q%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by predam** [comment]
I have the same issue. 
@testerez, can you make the codesandbox project public?

**Reply by testerez** [reply]
I updated the link. Let me know if you can't access.

**Reply by predam** [reply]
Yeah, so I can reproduce that in one of my projects. I delayed digging into it and came up with a hacky solution, enabling the query only on the client side.

Something like this:
```typescript
const [enabled, setEnabled] = useState(false);
const {data} = trpc.example.hello.useQuery({..}, {suspense: true, enabled})
useEffect(() => {
    setEnabled(true);
  }, []);
```

I am very interested in an actual fix.

**Comment by testerez** [comment]
For info, is looks like it's more a React behaviour. More infos here: https://github.com/facebook/react/issues/27420
